lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +30,115 @@ const _changesRequiredOutput = """
 Then, review the the changes, address any FIXMEs, and commit.
 """;
 
+String react16CodemodName = 'React16 Codemod';
+
+YamlMap pubspecYaml;
+
+Map<String, List<VersionConstraint>> packagesToCheckFor = {
+  'react': [
+    VersionConstraint.parse('^4.0.0'),
+    VersionConstraint.parse('^5.0.0')
+  ],
+  'over_react': [
+    VersionConstraint.parse('^2.0.0'),
+    VersionConstraint.parse('^3.0.0')
+  ]
+};
+
+VersionConstraint getDependencyVersion(String packageName) {
+  try {
+    if (pubspecYaml.containsKey('dependencies') &&
+        pubspecYaml['dependencies'].containsKey(packageName)) {
+      return VersionConstraint.parse(pubspecYaml['dependencies'][packageName]);
+    } else if (pubspecYaml.containsKey('dev_dependencies') &&
+        pubspecYaml['dev_dependencies'].containsKey(packageName)) {
+      return VersionConstraint.parse(
+          pubspecYaml['dev_dependencies'][packageName]);
+    }
+  } catch (e) {
+    return null;
+  }
+  return null;
+}
+
 void main(List<String> args) {
-  final query = FileQuery.dir(
-    pathFilter: isDartFile,
-    recursive: true,
-  );
-  exitCode = runInteractiveCodemodSequence(
-    query,
-    [
-      ReactDomRenderMigrator(),
-      ReactStyleMapsUpdater(),
-    ],
-    args: args,
-    defaultYes: true,
-    changesRequiredOutput: _changesRequiredOutput,
-  );
-
-  final logger = Logger('over_react_codemod.fixmes');
-  for (var dartFile in query.generateFilePaths()) {
-    final dartSource = File(dartFile).readAsStringSync();
-    if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
-      logger.severe(
-          'over_react_codemod validation comments are unaddressed in $dartFile');
-      exitCode = 1;
+  Map<String, bool> inTransition = {};
+  bool foundReactOrOverReact = false;
+
+  final react16CodemodLogger = Logger('over_react_codemod.react16');
+  react16CodemodLogger.onRecord.listen((LogRecord record) {
+    print(record.message);
+  });
+  react16CodemodLogger
+      .info('Checking if project needs to run $react16CodemodName...');
+  try {
+    pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  } catch (e) {
+    if (e is FileSystemException) {
+      react16CodemodLogger
+          .warning('Could not find pubspec.yaml, exiting codemod.');
+    } else if (e is YamlException || e is ArgumentError) {
```

Comment:
```
dynamic loadYaml(String yaml, {dynamic sourceUrl})

Loads a single document from a YAML string.

If the string contains more than one document, this throws a [YamlException]. In future releases, this will become an [ArgumentError].
```

Reply:
> If the string contains more than one document

Of course... I hate YAML so much haha.